### PR TITLE
Allow add-ons with long GUIDs with the allow-long-addon-guid waffle switch (bug 1203915)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1203915](https://bugzilla.mozilla.org/show_bug.cgi?id=1203915)

### CODE REVIEW

**Comment by magopian** [comment]
@kmaglione r?

**Comment by magopian** [comment]
Maybe someone else can r? @mstriemer @muffinresearch @kumar303

File: apps/files/utils.py [hunk 1]
```diff
@@ -459,7 +460,7 @@ def check_xpi_info(xpi_info, addon=None):
     guid = xpi_info['guid']
     if not guid:
         raise forms.ValidationError(_("Could not find an add-on ID."))
-    if len(guid) > 64:
+    if not waffle.switch_is_active('allow-long-addon-guid') and len(guid) > 64:
```
Comment (muffinresearch) [inline apps/files/utils.py]: Presumably there's still some upper limit?

Comment (magopian) [inline apps/files/utils.py]: The one enforced by the charfield (255) I believe

File: migrations/828-allow-long-addon-guid-switch.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO
+    waffle_switch (name, active, created, modified, note)
+VALUES
+    ('allow-long-addon-guid', 0, NOW(), NOW(), 'Allow submission of add-ons with a GUID longer than 64 chars (bug 1203915)');
```
Comment (kumar303) [inline migrations/828-allow-long-addon-guid-switch.sql]: :+1: for putting the bug number in the switch description

**Comment by kumar303** [comment]
I think testing `check_xpi_info` directly is the right approach but it would also be good to patch that method and add a test for `NewAddonForm` that makes sure it gets called. This will prevent future refactoring from removing that check accidentally and would be especially useful since `check_xpi_info` is buried under a couple other wrapper functions with some complex conditionals.

r+wc

**Comment by kumar303** [comment]
r+

**Pull request merged**